log/stat/stat.go

```diff
@@ -0,0 +1,107 @@
+package stat
+
+import (
+	"bytes"
+	"fmt"
+	"sync"
+	"text/tabwriter"
+
+	"github.com/peak/s5cmd/strutil"
+)
+
+const (
+	totalCount = iota
+	succCount
+)
+
+var (
+	enabled bool
+	stats   statistics
+)
+
+type statistics [2]syncMap
+
+// InitStat initializes collecting program statistics.
+func InitStat() {
+	enabled = true
+	for i := range stats {
+		stats[i] = syncMap{
+			Mutex:       sync.Mutex{},
+			mapStrInt64: map[string]int64{},
+		}
+	}
+}
+
+// syncMap is a statically typed and synchronized map.
+type syncMap struct {
+	sync.Mutex
+	mapStrInt64 map[string]int64
+}
+
+func (s *syncMap) add(key string, val int64) {
+	s.Lock()
+	defer s.Unlock()
+
+	s.mapStrInt64[key] += val
+}
+
+// Stat is for storing a particular statistics.
+type Stat struct {
+	Visited    string `json:"visited"`
+	SuccVisits int64  `json:"success visits"`
+	ErrVisits  int64  `json:"error visits"`
+}
+
+// Collect collects function execution data.
+func Collect(path string, err *error) func() {
```

Comment:
Why not `Collect(path string, err error)`? Can you state why we need to return function and use pointer error?

Reply:
You can ignore it. I forgot deferred calls :)